### PR TITLE
Mark completed for compaction when segment has reference lock

### PR DESCRIPTION
**Pull request by xiaocai2333**
issue: #17725 
Signed-off-by: Cai.Zhang <cai.zhang@zilliz.com>

### CODE REVIEW

File: internal/datacoord/meta.go [hunk 1]
```diff
@@ -780,15 +780,15 @@ func (m *meta) SetSegmentCompacting(segmentID UniqueID, compacting bool) {
 }
 
 func (m *meta) CompleteMergeCompaction(compactionLogs []*datapb.CompactionSegmentBinlogs, result *datapb.CompactionResult,
-	canCompaction func(segment *datapb.CompactionSegmentBinlogs) bool) error {
+	canCompaction func(segment *datapb.CompactionSegmentBinlogs) bool) (bool, error) {
```
Comment (jiaoew1991) [inline internal/datacoord/meta.go:783]: return bool and error at same time is confusing. maybe at L791 return a special error and let the caller handle the error is better

Comment (xiaocai2333) [reply internal/datacoord/meta.go:783]: good idea, I will do it.

Comment (xiaocai2333) [reply internal/datacoord/meta.go]: I have fixed it, please review @jiaoew1991

**Comment by xiaocai2333** [comment]
rerun ut

**Comment by xiaocai2333** [comment]
Need more detailed discussion.
/close

**Pull request closed without merging**